3DCalendar/ViewController.swift

```diff
@@ -72,4 +90,79 @@ class ViewController: UIViewController, ARSCNViewDelegate {
         // Reset tracking and/or remove existing anchors if consistent tracking is required
         
     }
+    
+    //pentagon via Bezier Path
+    func createPentagon() -> SCNNode {
+        let path = UIBezierPath()
+        path.move(to: CGPoint(x: 0.0, y: 0.6)) //A
+        path.addLine(to: CGPoint(x: -0.3, y: 0.4))//B
+        path.addLine(to: CGPoint(x: -0.2, y: 0.1))//C
+        path.addLine(to: CGPoint(x: 0.2, y: 0.1))//D
+        path.addLine(to: CGPoint(x: 0.3, y: 0.4))//E
+        path.close()
+        
+        let shape = SCNShape(path: path, extrusionDepth: 0.05) //20 cm
+        let color = UIColor.red
+        shape.firstMaterial?.diffuse.contents = color
+        shape.chamferRadius = 0.0
+        //create a node
+        let pentagonNode = SCNNode(geometry: shape)
+        //position the node and add to the scene
+        pentagonNode.position.z = -1
+        return pentagonNode
+    }
+
+    func createUIViewOnNode(vector: SCNVector3) -> SCNNode {
+        let dodecahedronNode = shapeNode()
+        dodecahedronNode.position.y = -10.0
+  
+        for item in 0...11 {
+            let material = SCNMaterial()
+            material.diffuse.contents = UIImage(view: calendarViewControllers[item].view)
+            dodecahedronNode.childNodes[item].geometry?.firstMaterial = material
+        }
+        
+        //7. Add To The Scene & Position It
+        sceneView.scene.rootNode.addChildNode(dodecahedronNode)
+        dodecahedronNode.position = vector
+        return dodecahedronNode
+    }
+    
+    func addRotationGesture() {
+        let panGesture = UIPanGestureRecognizer(target: self, action: #selector(handleGesture(_:)))
+        self.view.addGestureRecognizer(panGesture)
+        
+
+    }
+    
+    //gesture code based of off https://github.com/anoop4real/AR-RotateEarthWithGesture
+    @objc func handleGesture(_ gestureRecognizer: UIPanGestureRecognizer) {
+        
+        let translation = gestureRecognizer.translation(in: gestureRecognizer.view!)
+        
+        let x = Float(translation.y)
+        let y = Float(-translation.x)
+        let anglePan = (sqrt(pow(x,2)+pow(y,2)))*(Float)(Double.pi)/180.0
+        
+        var rotationVector = SCNVector4()
+        rotationVector.x = x
+        rotationVector.y = y
+        rotationVector.z = 0.0
+        rotationVector.w = anglePan
+        
+        
+        self.dodecaNode.rotation = rotationVector
+    }
+    
+}
+
+
+extension UIImage {
+    convenience init(view: UIView) {
```

Comment:
Let's verify if this conversion to an image thing is still necessary

Reply:
We can get rid of it, just set the diffuse contents = view